sw/device/tests/aon_timer_irq_test.c

```diff
@@ -86,23 +86,38 @@ static uint64_t tick_count_get(void) {
 }
 
 /**
- * Execute the wake up interrupt test.
+ * Execute the aon timer interrupt test.
  */
-static void execute_test(dif_aon_timer_t *aon_timer, uint32_t sleep_time_us) {
-  peripheral = kTopEarlgreyPlicPeripheralUnknown;
-  irq = kDifAonTimerIrqWdogTimerBark;
-  // The wake up time should be `sleepTime_us ±5%`.
-  uint32_t variation = sleep_time_us * 5 / 100;
+static void execute_test(dif_aon_timer_t *aon_timer, uint32_t irq_time_us,
+                         dif_aon_timer_irq_t expected_irq) {
+  // The interrupt time should be `irq_time_us ±5%`.
+  uint32_t variation = irq_time_us * 5 / 100;
   CHECK(variation > 0);
-  uint32_t sleep_range_h = sleep_time_us + variation;
-  uint32_t sleep_range_l = sleep_time_us - variation;
+  uint32_t sleep_range_h = irq_time_us + variation;
+  uint32_t sleep_range_l = irq_time_us - variation;
 
-  // Setup the timer and wait for the IRQ.
-  uint32_t sleep_cycles = (sleep_time_us * kClockFreqAonHz / 1000000);
-  LOG_INFO("Setting wakeup interrupt for %u us (%u cycles)", sleep_time_us,
-           sleep_cycles);
+  uint32_t count_cycles = (irq_time_us * kClockFreqAonHz / 1000000);
```

Comment:
How do you know this doesn't overflow?

Reply:
I changed to perform the computation in 64bits and add a check to verify overflow of 32 bits.

Reply:
There is not much value in testing a timeout over a couple of ms. AON timer is already verified in its own testbench.

Reply:
@sriyerg just to clarify

> There is not much value in testing a timeout over a couple of ms. AON timer is already verified in its own testbench.

Do you mean that the checks that test whether the timer interrupt occurred within a reasonable time frame (see code snippet below) are not useful, as this is covered elsewhere? 

```
  CHECK(time_elapsed < sleep_range_h && time_elapsed > sleep_range_l,
        "Timer took %u usec which is not in the range %u usec and %u usec",
        (uint32_t)time_elapsed, sleep_range_l, sleep_range_h);
```

If the above is correct interpretation of your comment, then I think it is at least useful to document this in the test. Something along the lines of "we don't need to worry about it, as it is covered by x,y,z".

Reply:
No I mean the `irq_time_us` input to this task. It doesn't have to be over 15-20 us for pre-Si testing. The check for interrupt occurring within a reasonable timeframe on the other hand, is very useful and necessary check.